plugins/spark/src/main/java/org/radargun/service/AbstractSparkLifecycle.java

```diff
@@ -0,0 +1,90 @@
+package org.radargun.service;
+
+import org.radargun.utils.Utils;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.regex.Pattern;
+
+/**
+ * Parent for Spark's Master & Worker lifecycles
+ *
+ * @author Matej Cimbora
+ */
+public abstract class AbstractSparkLifecycle extends ProcessLifecycle<AbstractSparkService> {
+
+   public AbstractSparkLifecycle(AbstractSparkService service) {
+      super(service);
+   }
+
+   protected void startInternal() {
+      super.startInternal();
+      service.registerAction(getStartPattern(), m -> {
+         for (int i = 0; i < 10; i++) {
+            String matchedString = m.group();
+            String fileName = matchedString.substring(matchedString.lastIndexOf("/"));
+            try {
+               FileInputStream fileInputStream = new FileInputStream(Paths.get(service.home, "logs", fileName).toFile());
+               getOutputReader().setStream(fileInputStream);
+               return;
+            } catch (FileNotFoundException e) {
+               log.error("Log file not created yet");
+               Utils.sleep(1000);
+               continue;
+            }
+         }
+         throw new IllegalStateException("Failed to attach reader to log file");
+      });
+   }
+
+   @Override
+   protected synchronized StreamReader getOutputReader() {
+      if (outputReader == null) {
+         outputReader = new FileOutputReader(line -> service.reportOutput(line));
+      }
+      return outputReader;
+   }
+
+   @Override
+   protected synchronized StreamReader getErrorReader() {
+      if (errorReader == null) {
+         errorReader = new FileOutputReader(line -> service.reportError(line));
+      }
+      return errorReader;
+   }
+
+   private static class FileOutputReader extends ProcessOutputReader {
```

Comment:
It would be even better if you could configure Spark to log to stdout rather than looking up file...